galaxy_importer/loaders.py

```diff
@@ -80,44 +108,83 @@ def _validate_name(self):
     def _log_loading(self):
         self.log.info(' ')
         self.log.info(
-            f'===== LOADING {self.content_type.name}: {self.name} =====')
+            f'===== LOADING {self.content_type.name}: {self.path_name} =====')
 
 
 class PluginLoader(ContentLoader):
     def load(self):
         self._log_loading()
-        self.doc_strings = self._get_doc_strings()
+        doc_strings = DocStringLoader(
+            content_type=self.content_type.value,
+            path=self._get_tmp_dir(self.root),
+            fq_name=self._get_fq_name(self.root, self.path_name),
+            logger=self.log,
+        ).load()
 
         return schema.Content(
-            name=self.name,
+            name=self.path_name,
             content_type=self.content_type,
-            doc_strings=self.doc_strings,
+            doc_strings=doc_strings,
         )
 
-    def _make_name(self):
-        return os.path.splitext(os.path.basename(self.rel_path))[0]
+    @staticmethod
+    def _make_name(rel_path):
+        return os.path.splitext(os.path.basename(rel_path))[0]
+
+    @staticmethod
+    def _make_path_name(rel_path, name):
+        dirname_parts = Path(os.path.dirname(rel_path)).parts[2:]
+        return '.'.join(list(dirname_parts) + [name])
+
 
-    def _get_doc_strings(self):
-        if self.content_type.value not in ANSIBLE_DOC_SUPPORTED_TYPES:
+class DocStringLoader():
+    def __init__(self, content_type, path, fq_name, logger=None):
+        self.content_type = content_type
+        self.path = path
+        self.fq_name = fq_name
+        self.log = logger or default_logger
+
+    def load(self):
+        if self.content_type not in ANSIBLE_DOC_SUPPORTED_TYPES:
             return None
 
         self.log.info('Getting doc strings via ansible-doc')
         json_output = self._run_ansible_doc()
 
         if not json_output:
             return
-        data = json.loads(json_output)
-        if self.name not in data.keys():
-            self.log.error(f'No "{self.name}" key in ansible-doc output')
-            return None
 
+        data = json.loads(json_output)
+        if not isinstance(data, dict):
+            raise exc.ImporterError('ansible-doc output not dictionary as expected')
+        if len(data.keys()) != 1:
+            raise exc.ImporterError('ansible-doc output did not return single top-level key')
+        data = list(data.values())[0]
         data = self._transform_doc_strings(data)
+
         return {
-            key: data[self.name].get(key, None)
+            key: data.get(key, None)
             for key in ANSIBLE_DOC_KEYS
         }
 
-    def _transform_doc_strings(self, data):
+    def _run_ansible_doc(self):
+        cmd = [
+            'env', f'ANSIBLE_COLLECTIONS_PATHS={self.path}',
+            'ansible-doc',
```

Comment:
Do we know the full path to where ansible-doc will be installed for the workers?  If so, specifying that here may eliminate some ambiguity.

Reply:
within `galaxy-dev` it should be in `/venv/bin/ansible-doc`, but `galaxy-importer` is called by `pulp-ansible` so in non-galaxy setups we won't know. also to keep supporting running from the commandline with `.main` for testing purposes we also won't know the binary path